src/ctl.c

```diff
@@ -2983,6 +2985,55 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	char *name = NULL;
+
+	READ_XOR_WRITE();
+
+	malloc_mutex_lock(tsd_tsdn(tsd), &ctl_mtx);
+	MIB_UNSIGNED(arena_ind, 1);
+	if (arena_ind == MALLCTL_ARENAS_ALL || arena_ind >=
+	    ctl_arenas->narenas) {
+		ret = EINVAL;
+		goto label_return;
+	}
+	arena_t *arena = arena_get(tsd_tsdn(tsd), arena_ind, false);
+	if (arena == NULL) {
+		ret = EFAULT;
+		goto label_return;
+	}
+	if (oldp != NULL && oldlenp != NULL) {
+		/*
+		 * Read the arena name.  When reading, the input oldp should point to
+		 * an array with a length no shorter than ARENA_NAME_LEN or the length
+		 * of the name.
+		 */
+		if (*oldlenp != sizeof(char *)) {
+			ret = EINVAL;
+			goto label_return;
+		}
+		name = *(char **)oldp;
+		arena_name_get(arena, name);
+		ret = 0;
+	} else {
+		/* Write the arena name. */
+		WRITE(name, char *);
+		if (name == NULL) {
+			ret = EINVAL;
+			goto label_return;
+		}
+		arena_name_set(arena, name);
+		ret = 0;
```

Comment:
This is shared by the two branches. Move it out to after the branch.

Reply:
Moved out. Besides, after changing the branch a bit, the neither read nor write situation will also reach the line. I assume that is what we want.